scripts/process/process.utils.R

```diff
@@ -0,0 +1,41 @@
+create_colors <- function(n, palette = "Blues"){
+  RColorBrewer::brewer.pal(n, palette)
+}
+
+create_breaks <- function(nbins, stepsize){
+  seq(from = 0, to = stepsize * (nbins - 1), length.out = nbins)
+}
+
+#' calculate cumulative precipitation and store as `precipVal`
+#' 
+#' this function assumes column names `DateTime`, 
+#' `id` (location of precip summarization), and `precipVal`
+#' 
+calc_cumulative_precip <- function(precip_df){
+  
+  precip_grp <- dplyr::group_by(precip_df, id)
+  precip_english <- dplyr::mutate(precip_grp, precipVal = precipVal/25.4) #convert mm to inches
+  precip_cumulative <- dplyr::mutate(precip_english, summ = cumsum(precipVal))
+  precip_data <- dplyr::select(precip_cumulative, DateTime, id, precipVal = summ)
+  
+  return(precip_data)
+}
+
+#' add a column to precip data based on the color bin  the precip
+#' value corresponds to
+#' 
+#' this function assumes column names `DateTime`, 
+#' `id` (location of precip summarization), and `precipVal`
+#' 
+bin_precip <- function(precip_df, breaks){
+  
+  precip_df <- dplyr::mutate(precip_df, cols = cut(precipVal, breaks = breaks, labels = FALSE)) 
```

Comment:
dplyr seems to me like a great candidate for the `@import` tag, despite the general encouragement to use `::`. you lose so much of the simplicity offered by dplyr when you take away pipes and have `::` in every line.

Reply:
I agree -- using `%>%` is one of the main reasons I turn to dplyr

Reply:
I think this was at my suggestion. I agree that importing dplyr is the way to go, but I also REALLY like to have the package called out explicitly this code is so collaborative, that being super clear about what functions are what is super helpful. That said, dplyr is becoming such a core package, maybe the :: rule could be overlooked in this case.

Reply:
I don't mind `::` but I also want access to the pipes. What if we did both:

```
library(dplyr)
dplyr::mutate(df, new_col = 5*old_col) %>% 
    dplyr::select(new_col) 
```

Reply:
I appreciate the need to be clear about where functions come from. You could even clarify the origin of `%>%` with this line:
```
`%>%` <- magrittr::`%>%`
```
which would permit you to keep omitting the `library(dplyr)` call while also using pipes in the code that follows.